nomos-libp2p/src/behaviour/nat/inner.rs

```diff
@@ -269,3 +269,207 @@ where
         Poll::Pending
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{cell::RefCell, pin::Pin};
+
+    use libp2p::{
+        autonat::v2::server,
+        core::{
+            transport::{DialOpts, ListenerId, MemoryTransport, Transport, TransportError},
+            upgrade::Version,
+        },
+        identify,
+        identity::Keypair,
+        plaintext,
+        swarm::{NetworkBehaviour, SwarmEvent},
+        yamux,
+    };
+    use libp2p_swarm_test::SwarmExt as _;
+    use rand::{
+        rngs::{OsRng, StdRng},
+        SeedableRng as _,
+    };
+
+    use super::*;
+    use crate::{
+        behaviour::nat::address_mapper::errors::AddressMapperError, config::NatMappingSettings,
+    };
+
+    thread_local! {
+        static MAPPER_CALLED: RefCell<bool> = const { RefCell::new(false) };
+    }
+
+    #[tokio::test]
+    async fn test_nat_mapping_happy_flow() {
```

Comment:
No need to prefix test cases with `test`. There's actually a lint for it which we will enable, at some point :)

Reply:
At this point I think we could add a bunch of other tests? Like what happens if the server does not support identify? Or if the server does not support Autonat? I don't think it takes too much time to add those tests but it would give us an idea of exactly what errors to expect on our nodes?